docs/design/rosbag2_snapshot_recorder/rosbag2_snapshot_recorder.md

```diff
@@ -0,0 +1,738 @@
+# Rosbag2 Snapshot
+
+## Context
+
+The following design handles functionality of keeping data from topics in buffer of specified size or from specific time duration and saving them after trigger.
+
+Terms:
+* "Snapshot" - save buffered data from topics to disk as rosbag
+* "Trigger" - an invocation letting a node know when to save rosbag to disk
+* "Recorder" - rosbag2 node which records data right after program start until killed
+
+## Feature description
+
+Rosbag in ROS1 had functionality of creating a [snapshot](https://github.com/ros/rosbag_snapshot). Snapshotter was a program which subscribed to topics and buffered messages until a `trigger` for snapshot was sent. As in Rosbag2 we want to support the following features:
+
+* Save data to rosbag on trigger
+  * Enable triggering buffer save to disk with service call
+
+* Pause and resume buffering process
```

Comment:
I don't understand this feature very well - my understanding for the intention of this feature is:

* rosbag2 is always keeping a circular buffer of the latest information - but not writing anything to disk
* when we trigger a snapshot - dump the entire buffer to disk

Why does a user want to pause buffering? This sounds to me like it makes rosbag2 just drop data, such that you wouldn't be able to take a snapshot of it.